### ISSUE TITLE
`pnpm list` throws an error with `file:` dependencies

### ISSUE THREAD

**Issue by lafkpages** [pnpm/pnpm#6914]
### pnpm version:
8.6.11

### Code to reproduce the issue:
My `package.json`:
```json
{
  "name": "gdr",
  "version": "0.0.1",
  "scripts": {
    ...
  },
  "devDependencies": {
    ...
  },
  "optionalDependencies": {
    "gdr": "file:wasm/pkg"
  },
  "type": "module"
}
```

### Expected behavior:
That I can see my installed packages and their versions via `pnpm list`.

### Actual behavior:
It throws an error:
```js
luis@metadata gdr % pnpm list          
 ERROR  Cannot read properties of undefined (reading 'startsWith')

pnpm: Cannot read properties of undefined (reading 'startsWith')
    at getPkgInfo (/Users/luis/.nvm/versions/node/v20.4.0/lib/node_modules/pnpm/dist/pnpm.cjs:116081:20)
    at /Users/luis/.nvm/versions/node/v20.4.0/lib/node_modules/pnpm/dist/pnpm.cjs:116477:59
    at Array.forEach (<anonymous>)
    at dependenciesHierarchyForPackage (/Users/luis/.nvm/versions/node/v20.4.0/lib/node_modules/pnpm/dist/pnpm.cjs:116476:33)
    at async /Users/luis/.nvm/versions/node/v20.4.0/lib/node_modules/pnpm/dist/pnpm.cjs:116440:11
    at async Promise.all (index 0)
    at async buildDependenciesHierarchy (/Users/luis/.nvm/versions/node/v20.4.0/lib/node_modules/pnpm/dist/pnpm.cjs:116437:8)
    at async list (/Users/luis/.nvm/versions/node/v20.4.0/lib/node_modules/pnpm/dist/pnpm.cjs:117059:16)
    at async /Users/luis/.nvm/versions/node/v20.4.0/lib/node_modules/pnpm/dist/pnpm.cjs:219737:21
    at async main (/Users/luis/.nvm/versions/node/v20.4.0/lib/node_modules/pnpm/dist/pnpm.cjs:219704:34)
luis@metadata gdr % 
```

### Additional information:

 - `node -v` prints: v20.4.0
 - Windows, macOS, or Linux?: MacOS

**Comment by zkochan** [comment]
It was fixed by https://github.com/pnpm/pnpm/pull/6911 but not released yet

**Comment by lafkpages** [comment]
I thought I searched for duplicates before posting, I guess I didn't search enough. Thanks

**Comment by zkochan** [comment]
The fix has been published with v8.6.12

**Comment by lafkpages** [comment]
Thanks!
